samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -903,14 +903,22 @@ class SamzaContainer(
       info("Starting diagnostics.")
 
       try {
-        val diagnosticsAppender = Class.forName(config.getDiagnosticsAppenderClass).
-          getDeclaredConstructor(classOf[SamzaContainerMetrics]).newInstance(this.metrics);
+        var diagnosticsAppender = Util.getObj("org.apache.samza.logging.log4j.SimpleDiagnosticsAppender", (classOf[SamzaContainerMetrics], this.metrics))
+        info("Attached log4j diagnostics appender.")
       }
       catch {
         case e@(_: ClassNotFoundException | _: InstantiationException | _: InvocationTargetException) => {
-          error("Failed to instantiate diagnostic appender", e)
-          throw new ConfigException("Failed to instantiate diagnostic appender class " +
-            config.getDiagnosticsAppenderClass, e)
+          try {
+            val diagnosticsAppender = Util.getObj("org.apache.samza.logging.log4j2.SimpleDiagnosticsAppender", (classOf[SamzaContainerMetrics], this.metrics))
+            info("Attached log4j2 diagnostics appender.")
+          } catch {
+            case e : Throwable => {
+              warn("Failed to instantiate neither diagnostic appender for sending error information to diagnostics stream", e)
```

Comment:
Should not ignore throwables. Maybe catch specific exceptions / Exception?

Reply:
Same for the catch below.

Reply:
I dont want the uncaught exception handler to trigger container failure, if diagnostics cannot be started.

Reply:
Yeah, but I think you can catch the specific exceptions / Exception. E.g., all three exceptions above extend from Exception.